### ISSUE TITLE
[nre] `captures[i]` should return `Option[string]` instead of recent subtly-breaking change that throws

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#9941]
/cc @flaviut 
that's not what I meant, but I worded it poorly; 
after this recent PR https://github.com/nim-lang/Nim/pull/9755 some of my code broke, but it broke at RT instead of CT, which is not as good as if it failed at CT (because no-one has 100% test coverage, and who knows what other projects would break in the wild...)

how about the following:
* `m.captures[0]` returns the standard `Option[string]` instead of this non-standard `[]` that returns a `string` but throws on un-captured group?
that would've forced me to fix my code and guaranteed it would work 100% ; right now I have no such warrantee until my code hits such an issue in the wild

* furthermore, with `Option[string]`, I can also write (if my application doesn't care to distinguish bw nil vs empty string):
```
m.captures[0].get("") # uses `proc get[T](self: Option[T]; otherwise: T): T`
```
basically, things are more composable; with current API, it's written more awkwardly.

## Alternative (for a smoother transition via deprecation path compared to this PR)
keep the old `captures` (that returns `""` instead of `nil` for no capture) but mark it as deprecated, and add a new `capturesOption` (or whatever name) that returns an `Option[string]`

## Note
`Option[string]` would just make everything more consistent, eg with `captures.toSeq`and `captures.toTable` that must return `seq[Option[string]]` anyway /cc @dom96

**Comment by Araq** [comment]
I don't like ``Option[string]`` at all and yet I agree completely with this proposal. :-)

**Comment by dom96** [comment]
once again, `nre` is scheduled to be removed. Let's stop making proposals for it and move it to a nimble package.

**Comment by timotheecour** [comment]
`nre` is better in almost all respects compared to `re`, both in terms of API (usability etc) as well as features.
It makes makes sense since it was built after lessons were learned from `re`.

The few features found in `re` but not `nre` are "easily" (YMMV) fixable without breaking changes, the other way around not at all (would definitely involve heavy breaking changes). These are tracked here: https://github.com/nim-lang/Nim/issues/8150

So if 1 package is to be removed before v1 (say graveyard or nimble package), it should be `re`.

see this summary from @flaviut 
https://github.com/nim-lang/Nim/issues/7278#issuecomment-427682084

**Comment by Araq** [comment]
It is better for *you*, my code all uses ``re`` and I don't want to rewrite working code all day long just because of foolish politics.
